analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -158,7 +162,31 @@ private fun SpdxExternalDocumentReference.resolve(definitionFile: File, issues:
         }
     } else {
         tempDir = createOrtTempDir()
-        OkHttpClientHelper.downloadFile(uri.toString(), tempDir).getOrNull() ?: run {
+
+        val client = OkHttpClientHelper.buildClient {
+            addInterceptor { chain ->
```

Comment:
As this interceptor is quite generic, it could be worth extracting it to an utility class, e.g. `PreemptiveAuthenticationInterceptor`.

Reply:
I've extracted it to an extension function instead.